packages/agreement-process/src/services/agreementService.ts

```diff
@@ -1274,6 +1312,50 @@ export function agreementServiceBuilder(
 
       return updatedAgreement;
     },
+    async internalArchiveAgreementAfterDelegationRevocation(
+      agreementId: AgreementId,
+      delegationId: DelegationId,
+      correlationId: CorrelationId,
+      logger: Logger
+    ): Promise<void> {
+      logger.info(
+        `Archiving agreement ${agreementId} due to revocation of delegation ${delegationId}`
+      );
+
+      const agreement = await retrieveAgreement(agreementId, readModelService);
+
+      assertExpectedState(
+        agreementId,
+        agreement.data.state,
+        agreementArchivableStates
+      );
+
+      const activeConsumerDelegation = await retrieveConsumerDelegationById(
+        delegationId,
+        readModelService
+      );
+
+      const updatedAgreement: Agreement = {
+        ...agreement.data,
+        state: agreementState.archived,
+        stamps: {
+          ...agreement.data.stamps,
+          archiving: createStamp(undefined, {
+            consumerDelegation: activeConsumerDelegation,
+            producerDelegation: undefined,
+          }),
```

Comment:
As discussed in the comment, we probably don't need to set a stamp here (and we can remove the changes to the stamps creation function)